### ISSUE TITLE
Null reference exception in SimpleValueObject's implicit operator

### ISSUE THREAD

**Issue by lawrence-laz** [vkhorikov/CSharpFunctionalExtensions#289]
Hi, I've stumbled on a null reference exception when assigning a null simple value object and now I'm wondering whether this is intentional or just something that was missed. The reproduction is provided below.

[dotnet fiddle](https://dotnetfiddle.net/NBJzd7):
 ```csharp
using System.Collections.Generic;
using CSharpFunctionalExtensions;

#nullable enable

Foo? foo = default;
string? fooString = foo; // Throws null reference exception

public class Foo : SimpleValueObject<string>
{
	public Foo(string value) : base(value) 
	{
	}

	protected override IEnumerable<object> GetEqualityComponents()
    {
        yield return Value.ToLower();
    }
}
```
The root of exception comes from the [implicit operator](https://github.com/vkhorikov/CSharpFunctionalExtensions/blob/master/CSharpFunctionalExtensions/ValueObject/SimpleValueObject.cs#L26):
```csharp
public static implicit operator T(SimpleValueObject<T> valueObject) => valueObject.Value;
```
When `valueObject` is null the exception is thrown. I get it that usually you don't expect methods to tolerate calling them with `nulls`, but in case of implicit operators it's a bit surprising and inconvenient that code like this throws:
```csharp
Foo? foo = default;
string? fooString = foo;
```
This could be easily remedied by using a null propagating operator same as in `ToString()` method.
What are you thoughts?

**Comment by vkhorikov** [comment]
Yeah, it's a bug. I've pushed a fix, will be published soon with v2.15.3.

**Comment by lawrence-laz** [comment]
Great, it's resolved now 🙂
